src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/ModuleBuilderImpl.cs

```diff
@@ -399,6 +406,199 @@ private void WriteMethods(List<MethodBuilderImpl> methods, List<GenericTypeParam
+                int previousNonHiddenStartColumn = pair.Value[i].StartColumn;
+                pair = enumerator.Current;
+                spBlobBuilder.WriteCompressedInteger(0);
+                spBlobBuilder.WriteCompressedInteger(MetadataTokens.GetRowNumber(GetDocument(pair.Key)));
+                PopulateSequencePointsBlob(spBlobBuilder, pair.Value, previousNonHiddenStartLine, previousNonHiddenStartColumn);
+            }
+
+            return _pdbBuilder.GetOrAddBlob(spBlobBuilder);
+        }
+
+        private static void PopulateSequencePointsBlob(BlobBuilder spBlobBuilder, List<SequencePoint> sequencePoints, int previousNonHiddenStartLine = -1, int previousNonHiddenStartColumn = -1)
+        {
+            for (int i = 0; i < sequencePoints.Count; i++)
+            {
+                // IL offset delta:
+                if (i > 0)
+                {
+                    spBlobBuilder.WriteCompressedInteger(sequencePoints[i].Offset - sequencePoints[i - 1].Offset);
+                }
+                else
+                {
+                    spBlobBuilder.WriteCompressedInteger(sequencePoints[i].Offset);
+                }
+
+                if (sequencePoints[i].IsHidden)
+                {
+                    spBlobBuilder.WriteUInt16(0);
+                    continue;
+                }
+
+                // Delta Lines & Columns:
+                SerializeDeltaLinesAndColumns(spBlobBuilder, sequencePoints[i]);
+
+                // delta Start Lines & Columns:
+                if (previousNonHiddenStartLine < 0)
+                {
+                    Debug.Assert(previousNonHiddenStartColumn < 0);
+                    spBlobBuilder.WriteCompressedInteger(sequencePoints[i].StartLine);
+                    spBlobBuilder.WriteCompressedInteger(sequencePoints[i].StartColumn);
+                }
+                else
+                {
+                    spBlobBuilder.WriteCompressedSignedInteger(sequencePoints[i].StartLine - previousNonHiddenStartLine);
+                    spBlobBuilder.WriteCompressedSignedInteger(sequencePoints[i].StartColumn - previousNonHiddenStartColumn);
+                }
+
+                previousNonHiddenStartLine = sequencePoints[i].StartLine;
+                previousNonHiddenStartColumn = sequencePoints[i].StartColumn;
+            }
+        }
+
+        private void AddLocalScope(MethodDefinitionHandle methodHandle, ImportScopeHandle parentImport, LocalVariableHandle firstLocalVariableHandle, Scope scope)
+        {
+            parentImport = GetImportScopeHandle(scope._importNamespaces, parentImport);
+            firstLocalVariableHandle = GetLocalVariableHandle(scope._locals, firstLocalVariableHandle);
+            _pdbBuilder.AddLocalScope(methodHandle, parentImport, firstLocalVariableHandle,
+                constantList: MetadataTokens.LocalConstantHandle(1), scope._startOffset, scope._endOffset - scope._startOffset);
```

Comment:
Where is LocalConstantHandle(1) coming from?

Reply:
https://github.com/dotnet/runtime/blob/64a937cf26c5bac130d8b099e98940981e17aa19/src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/Ecma335/MetadataBuilder.Tables.cs#L1266